instrumentation/jsonrpc4j-1.3/library/src/main/java/io/opentelemetry/instrumentation/jsonrpc4j/v1_3/OpenTelemetryJsonRpcInvocationListener.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jsonrpc4j.v1_3;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.googlecode.jsonrpc4j.InvocationListener;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import java.lang.reflect.Method;
+import java.util.List;
+
+public final class OpenTelemetryJsonRpcInvocationListener implements InvocationListener {
+
+  private final Instrumenter<JsonRpcServerRequest, JsonRpcServerResponse> serverInstrumenter;
+
+  private static final ThreadLocal<Context> threadLocalContext = new ThreadLocal<>();
+  private static final ThreadLocal<Scope> threadLocalScope = new ThreadLocal<>();
+
+  public OpenTelemetryJsonRpcInvocationListener(
+      Instrumenter<JsonRpcServerRequest, JsonRpcServerResponse> serverInstrumenter) {
+    this.serverInstrumenter = serverInstrumenter;
+  }
+
+  /**
+   * This method will be invoked prior to a JSON-RPC service being invoked.
+   *
+   * @param method is the method that will be invoked.
+   * @param arguments are the arguments that will be passed to the method when it is invoked.
+   */
+  @Override
+  public void willInvoke(Method method, List<JsonNode> arguments) {
+    Context parentContext = Context.current();
+    JsonRpcServerRequest request = new JsonRpcServerRequest(method, arguments);
+    if (!serverInstrumenter.shouldStart(parentContext, request)) {
+      return;
+    }
+
+    Context context = serverInstrumenter.start(parentContext, request);
+    threadLocalContext.set(context);
```

Comment:
I'd use a single thread local that carries context, scope and request.

Reply:
that's better, updated